Development/boost/asio/ssl/use_tmp_ecdh.hpp

```diff
@@ -40,16 +45,19 @@ struct evp_pkey_cleanup
     ~evp_pkey_cleanup() { if (p) ::EVP_PKEY_free(p); }
 };
 
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
 struct ec_key_cleanup
```

Comment:
@lo-simon I can't remember, was this cleanup-struct idiom adopted from the Boost.ASIO code?
It serves the same purpose as your usage of `std::unique_ptr` in _nmos/jwk_utils.cpp_, is that right?

Reply:
yes, it is; see the following example extracted from asio\ssl\impl\contentext.ipp

```
struct context::bio_cleanup
{
  BIO* p;
  ~bio_cleanup() { if (p) ::BIO_free(p); }
};

struct context::x509_cleanup
{
  X509* p;
  ~x509_cleanup() { if (p) ::X509_free(p); }
};

struct context::evp_pkey_cleanup
{
  EVP_PKEY* p;
  ~evp_pkey_cleanup() { if (p) ::EVP_PKEY_free(p); }
};

#if (OPENSSL_VERSION_NUMBER < 0x30000000L)
struct context::rsa_cleanup
{
  RSA* p;
  ~rsa_cleanup() { if (p) ::RSA_free(p); }
};

struct context::dh_cleanup
{
  DH* p;
  ~dh_cleanup() { if (p) ::DH_free(p); }
};
#endif // (OPENSSL_VERSION_NUMBER < 0x30000000L)
```